src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java

```diff
@@ -1272,49 +1273,46 @@ public String getCriteriaSql(DrugExposure criteria)
     }
     
     return query;
-  }  
+  }
+
+  protected <T extends Criteria> String getCriteriaSql(BaseCriteriaSqlBuilder<T> builder, T criteria) {
+    String query = builder.getCriteriaSql(criteria);
+    return processCorrelatedCriteria(query, criteria);
+  }
   
   @Override
-  public String getCriteriaSql(Measurement criteria)
-  {
-    String query = new MeasurementSqlBuilder<>().getCriteriaSql(criteria);
-    return processCorrelatedCriteria(query, criteria);
+  public String getCriteriaSql(Measurement criteria) {
+    return getCriteriaSql(new MeasurementSqlBuilder<>(), criteria);
   }
   
   @Override
   public String getCriteriaSql(Observation criteria) {
-    String query = new ObservationSqlBuilder<>().getCriteriaSql(criteria);
-    return processCorrelatedCriteria(query, criteria);
+    return getCriteriaSql(new ObservationSqlBuilder<>(), criteria);
```

Comment:
I noticed that each of these getCriteriaSql create a new instance of the object for each invocation of the method.  Can they be allocated once and reused, or is it necessary that they get constructed for each invocation?

Reply:
@chrisknoll , addressed